commit afc888891e6a37fbe8ab4c280e96dbe098d7fbb1 (refs/changes/19/5419/2)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2019-01-31T10:40:56-07:00 (8 months ago)
    
    TRITON-1101 add disklayout support for large storage servers
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/scripts/joysetup.sh b/scripts/joysetup.sh
index 2b4e6cfb..1423a6dd 100755
--- a/scripts/joysetup.sh
+++ b/scripts/joysetup.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright 2017 Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 #
@@ -621,12 +621,16 @@ if [[ "$(zpool list)" == "no pools available" ]] \
     if [[ -n "${CONFIG_cache}" && "${CONFIG_cache}" == "false" ]]; then
         dl2="-c"
     fi
+    if [[ -n "${CONFIG_width}" ]]; then
+        dl3="-w"
+        dl4="${CONFIG_width}"
+    fi
     if [[ -n "${CONFIG_layout}" ]]; then
-        dl3="${CONFIG_layout}"
+        dl5="${CONFIG_layout}"
     fi
 
-    if ! /usr/bin/disklayout ${dl0} ${dl1:+"$dl1"} ${dl2} "${dl3}" \
-         > ${POOL_FILE}; then
+    if ! /usr/bin/disklayout ${dl0} ${dl1:+"$dl1"} ${dl2} \
+	${dl3} ${dl4:+"$dl4"} "${dl5}" > ${POOL_FILE}; then
         fatal "disk layout failed"
     fi
 
